### PR TITLE
Fix alps custom estimator

### PR DESCRIPTION
**Pull request by joyyoj**
Alps support custom estimator.

### CODE REVIEW

File: sql/codegen_alps.go [hunk 1]
```diff
@@ -43,6 +43,7 @@ type alpsFiller struct {
 	Y      string
 
 	// Train
+	ImportCode        string
```
Comment (typhoonzero) [inline sql/codegen_alps.go:46]: maybe the name `CustomEstimatorImportCode` make more sence?

File: sql/codegen_alps.go [hunk 2]
```diff
@@ -89,20 +100,28 @@ func newALPSTrainFiller(pr *extendedSelect, db *DB) (*alpsFiller, error) {
 
 	var odpsConfig = &gomaxcompute.Config{}
 	var columnInfo map[string]*columnSpec
-	var meta metadata
 
 	// TODO(joyyoj) read feature mapping table's name from table attributes.
 	// TODO(joyyoj) pr may contains partition.
 	fmap := featureMap{pr.tables[0] + "_feature_map", ""}
+	var meta metadata
+	selectFields := pr.standardSelect.fields
```
Comment (typhoonzero) [inline sql/codegen_alps.go]: What if `SELECT *`?

File: sql/codegen_alps.go [hunk 3]
```diff
@@ -111,7 +130,7 @@ func newALPSTrainFiller(pr *extendedSelect, db *DB) (*alpsFiller, error) {
 		}
 		meta.columnInfo = &columnInfo
 	}
-	fields := make([]string, 0) // TODO use complete fields
+	//fields := make([]string, 0) // TODO use complete fields
```
Comment (typhoonzero) [inline sql/codegen_alps.go]: please clean commented code

File: sql/codegen_alps.go (hunk 1 above)
Comment (joyyoj) [reply sql/codegen_alps.go:46]: I think ImportCode is better for it is more general. It's probable not limited to custom estimator imports In future.

File: sql/codegen_alps.go (hunk 2 above)
Comment (joyyoj) [reply sql/codegen_alps.go]: I know this problem. I try use 'desc table' clause to get fields but maxcompute report "not support instance type". I add a TODO here.

Comment (joyyoj) [reply sql/codegen_alps.go]: Use `select * from table limit 1 `clause to get fields. if table is not empty, it works

**Review by typhoonzero — APPROVED** [review]
LGTM++

File: sql/codegen_alps.go (hunk 2 above)
Comment (typhoonzero) [reply sql/codegen_alps.go]: Cool~

**Pull request merged**